pkg/agent/config.go

```diff
@@ -25,10 +31,72 @@ type Endpoint struct {
 }
 
 type dataGatherer struct {
-	Kind     string `yaml:"kind"`
-	Name     string `yaml:"name"`
-	DataPath string `yaml:"data-path,omitempty"`
-	Config   []byte `yaml:"config"`
+	Kind     string
+	Name     string
+	DataPath string
+	Config   datagatherer.Config
+}
+
+func reMarshal(rawConfig interface{}, config datagatherer.Config) error {
+	bb, err := yaml.Marshal(rawConfig)
+	if err != nil {
+		return nil
+	}
+
+	err = yaml.Unmarshal(bb, config)
+	if err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// UnmarshalYAML unmarshals a dataGatherer resolving the type according to Kind.
+func (dg *dataGatherer) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
This is called implicitly when loading yaml data into a dg right?

Reply:
Exactly: https://godoc.org/gopkg.in/yaml.v2#Unmarshaler

Reply:
:+1: thanks.